idasen/__init__.py

```diff
@@ -75,12 +80,26 @@ def __init__(self, mac: str):
         self._client = BleakClient(self._mac)
 
     async def __aenter__(self):
-        await self._client.__aenter__()
+        await self.connect()
         return self
 
     async def __aexit__(self, *args, **kwargs) -> Optional[bool]:
         return await self._client.__aexit__(*args, **kwargs)
 
+    async def connect(self):
```

Comment:
This should be private since `bleak` depends on the context manager for cleanup.
```suggestion
    async def _connect(self):
```